test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/SequenceMethodsTests.cs

```diff
@@ -0,0 +1,209 @@
+{
+    /// <summary>
+    /// Tests to check sequence methods support.
+    /// </summary>
+    public class SequenceMethodsTests
+    {
+        private readonly TestContext _ctx;
+
+        public SequenceMethodsTests()
+        {
+            _ctx = new TestContext();
+        }
+
+        [Fact]
+        public void Any()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count");
+            _ctx.InterceptRequestAndMockResponse("91");
+            bool exists1 = _ctx.Customers.Any();
+            Assert.True(exists1);
+        }
+
+        [Fact]
+        public void AnyPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'ab')");
+            _ctx.InterceptRequestAndMockResponse("6");
+            bool exists2 = _ctx.Customers.Any(c => c.Name.Contains("ab"));
+            Assert.True(exists2);
+
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'thisdoesntexist')");
+            _ctx.InterceptRequestAndMockResponse("0");
+            bool exists3 = _ctx.Customers.Any(c => c.Name.Contains("thisdoesntexist"));
+            Assert.False(exists3);
+        }
+
+        [Fact]
+        public void CountPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'ab')");
+            _ctx.InterceptRequestAndMockResponse("6");
+            int count1 = _ctx.Customers.Count(c => c.Name.Contains("ab"));
+            Assert.Equal(6, count1);
+        }
+
+        [Fact]
+        public void LongCountPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers/$count?$filter=contains(ContactName,'ab')");
+            _ctx.InterceptRequestAndMockResponse("6");
+            long count2 = _ctx.Customers.LongCount(c => c.Name.Contains("ab"));
+            Assert.Equal(6, count2);
+        }
+
+        [Fact]
+        public void FirstPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers?$filter=ContactName ne 'John'&$top=1");
+            _ctx.InterceptRequestAndMockResponseValue("Customers", "[{\"CustomerID\":\"ALFKI\",\"CompanyName\":\"Alfreds Futterkiste\",\"ContactName\":\"Maria Anders\",\"Address\":\"Obere Str. 57\",\"City\":\"Berlin\"}]");
+            Customer c1 = _ctx.Customers.First(c => c.Name != "John");
+            Assert.Equal("ALFKI", c1.Id);
+            Assert.Equal("Maria Anders", c1.Name);
+            Assert.Equal("Berlin", c1.City);
+        }
+
+        [Fact]
+        public void FirstOrDefaultPredicate()
+        {
+            _ctx.InterceptRequestAndAssertUri("/Customers?$filter=ContactName eq 'John'&$top=1");
+            _ctx.InterceptRequestAndMockResponseValue("Customers", "[]");
+            Customer c2 = _ctx.Customers.FirstOrDefault(c => c.Name == "John");
+            Assert.Null(c2);
+        }
+
+        [Fact]
+        public void SinglePredicate()
```

Comment:
What does `SinglePredicate` return when there's no match or when there are multiple matches? Could we test those paths as well?